tests/kernels/test_space_to_batch.cpp

```diff
@@ -0,0 +1,110 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "kernel_test.h"
+#include <gtest/gtest.h>
+#include <iostream>
+#include <nncase/kernels/stackvm/tensor_ops.h>
+#include <nncase/runtime/datatypes.h>
+#include <nncase/runtime/runtime_tensor.h>
+#include <nncase/runtime/simple_types.h>
+#include <nncase/runtime/stackvm/opcode.h>
+#include <ortki/operators.h>
+
+using namespace nncase;
+using namespace nncase::runtime;
+using namespace ortki;
+
+class SpaceToBatchTest : public KernelTest,
+                         public ::testing::TestWithParam<
+                             std::tuple<nncase::typecode_t, dims_t, dims_t>> {
+  public:
+    void SetUp() override {
+        auto &&[typecode, l_shape, r_shape] = GetParam();
+
+        lhs = hrt::create(typecode, l_shape, host_runtime_tensor::pool_cpu_only)
+                  .expect("create tensor failed");
+        init_tensor(lhs);
+
+        rhs = hrt::create(typecode, r_shape, host_runtime_tensor::pool_cpu_only)
+                  .expect("create tensor failed");
+        init_tensor(rhs);
+    }
+
+    void TearDown() override {}
+
+  protected:
+    runtime_tensor lhs;
+    runtime_tensor rhs;
+};
+
+INSTANTIATE_TEST_SUITE_P(SpaceToBatch, SpaceToBatchTest,
+                         testing::Combine(testing::Values(dt_float32, dt_int32,
+                                                          dt_int64),
+                                          testing::Values(dims_t{1, 3, 16, 16},
+                                                          /*dims_t { 3, 16, 16
+                                                          }, dims_t { 16, 16 },
+                                                          dims_t { 16 },*/
+                                                          dims_t{1}),
+                                          testing::Values(dims_t{1, 3, 16, 16},
+                                                          /*dims_t { 3, 16, 16
+                                                          }, dims_t { 16, 16 },
+                                                          dims_t { 16 },*/
+                                                          dims_t{1})));
+
+TEST_P(SpaceToBatchTest, SpaceToBatch) {
+    //    auto l_ort = runtime_tensor_2_ort_tensor(lhs);
+    //    auto r_ort = runtime_tensor_2_ort_tensor(rhs);
+
+    // expected
+    //    size_t size = 0;
+    float_t expected_array[] = {1, 3, 9,  11, 2, 4, 10, 12,
+                                5, 7, 13, 15, 6, 8, 14, 16};
```

Comment:
expected_array has nothing to do with lhs/rhs

Reply:
done